include/devicetree/zephyr.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Zephyr-specific devicetree /chosen properties
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_ZEPHYR_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_ZEPHYR_H_
+
+/**
+ * @defgroup devicetree-zephyr Zephyr's /chosen nodes
+ * @{
+ */
+
+/*
+ * This file is currently deliberately not defining macros for some
+ * existing zephyr,foo chosen nodes, such as zephyr,sram, until there
+ * are users for them. Feel free to extend it as needed.
+ */
+
+#if DT_HAS_CHOSEN(zephyr_flash)
+/** Node identifier for the /chosen node's zephyr,flash property value. */
+#define DT_ZEPHYR_FLASH DT_CHOSEN(zephyr_flash)
+/** The /chosen zephyr,flash node's label property. */
+#define DT_ZEPHYR_FLASH_LABEL DT_LABEL(DT_ZEPHYR_FLASH)
+#elif defined(DT_FLASH_DEV_NAME) /* fall back on dts_fixup.h */
+#define DT_ZEPHYR_FLASH_LABEL DT_FLASH_DEV_NAME
```

Comment:
I'd stick a __deprecated on this for now, we can remove right before release, but that will catch any PRs or in-tree usage.

Reply:
> I'd stick a __deprecated on this for now, we can remove right before release, but that will catch any PRs or in-tree usage.

I think you and @nashif ought to talk about this. From what I understood on Slack, he believes devicetree ought to be optional as a long-term thing. I'm confused about what's going on.

Reply:
Ok, not sure how that is related.

Reply:
How can we remove this if we always need to support non-DT ways to name this string?

Reply:
I.e. this is a specific instance of a more general situation where I am not sure if we're going to deprecate and remove non-devicetree ways of naming and creating devices, or if we need fallbacks and fixups forever.